Packs/FeedGithub/Integrations/FeedGithub/FeedGitHub.py

```diff
@@ -0,0 +1,603 @@
+import demistomock as demisto
+from CommonServerPython import *
+from TAXII2ApiModule import *
+import plyara
+import plyara.utils
+import tldextract
+
+CONTEXT_PREFIX = "GITHUB"
+RAW_RESPONSE = []
+
+
+class Client(BaseClient):
+    def __init__(self, base_url: str, verify: bool, proxy: bool, owner: str, repo: str, headers: dict):
+        base_url = urljoin(base_url, f"/repos/{owner}/{repo}")
+        super().__init__(base_url=base_url, verify=verify, proxy=proxy, headers=headers)
+
+    def get_commits_between_dates(self, since, until) -> list:
+        """
+        Retrieves the SHA of the base commit of the repository.
+
+        This function fetches all commits of the repository and returns the SHA of the latest commit,
+        which represents the base commit of the repository.
+
+        Returns:
+            str: The SHA of the base commit of the repository.
+        """
+        parsed_since = dateparser.parse(since)
+        parsed_until = dateparser.parse(until)
+        if not (parsed_since and parsed_until):
+            demisto.debug(f"Unable to parse date string: since:{since} until:{until}")
+            raise ValueError(f"Unable to parse date string: since:{since} until:{until}")
+        params = {
+            "since": parsed_since.isoformat(),
+            "until": parsed_until.isoformat(),
+        }
+        response = self._http_request("GET", full_url=f"{self._base_url}/commits", params=params, resp_type="response")
+        demisto.debug(f"The base get_base_head_commits_sha() raw response: {response}")
+        return [commit.get("sha") for commit in self._extract_commits(response)]
+
+    def _extract_commits(self, response) -> list:
+        all_commits = []
+        if "next" not in response.links:
+            all_commits.extend(response.json())
+        while "next" in response.links:
+            data = response.json()
+            all_commits.extend(data)
+            response = self._http_request("GET", full_url=response.links["next"]["url"], resp_type="response")
+        demisto.debug(f"list of all commits in the given time frame{all_commits}")
```

Comment:
we probably dont need to log the details of all commits. lets log the # instead

Reply:
I took the debugger out of here and put it inside the loop